test/Common/DurableOrchestrationClientBaseTests.cs

```diff
@@ -90,12 +99,72 @@ public async Task TerminateAsync_NonRunningOrchestrator_ThrowsException()
             orchestrationServiceClientMock.Setup(x => x.GetOrchestrationStateAsync(It.IsAny<string>(), It.IsAny<bool>()))
                 .ReturnsAsync(GetInstanceState(OrchestrationStatus.Completed));
-            var durableOrchestrationClient = (IDurableOrchestrationClient)new DurableOrchestrationClient(orchestrationServiceClientMock.Object, GetDurableTaskExtension(), new OrchestrationClientAttribute { });
+            var durableExtension = GetDurableTaskExtension();
+            var durableOrchestrationClient = (IDurableOrchestrationClient)new DurableOrchestrationClient(orchestrationServiceClientMock.Object, durableExtension, durableExtension.HttpApiHandler, new OrchestrationClientAttribute { });
 
             await Assert.ThrowsAnyAsync<InvalidOperationException>(async () => await durableOrchestrationClient.TerminateAsync("invalid_instance_id", "any reason"));
             orchestrationServiceClientMock.Verify(x => x.ForceTerminateTaskOrchestrationAsync(It.IsAny<string>(), It.IsAny<string>()), Times.Never());
         }
 
+        [Fact]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        public async Task HttpRequest_HttpRequestMessage_ClientMethods_Identical()
+        {
+            var instanceId = Guid.NewGuid().ToString();
+            var orchestrationServiceClientMock = new Mock<IOrchestrationServiceClient>(MockBehavior.Strict);
+            orchestrationServiceClientMock.Setup(x => x.GetOrchestrationStateAsync(It.IsAny<string>(), It.IsAny<bool>()))
+                .ReturnsAsync(GetInstanceState(OrchestrationStatus.Completed));
+            var durableExtension = GetDurableTaskExtension();
+            var httpHandler = new ExtendedHttpApiHandler(new Mock<IDurableOrchestrationClient>(MockBehavior.Strict).Object);
+            var durableOrchestrationClient = (IDurableOrchestrationClient)new DurableOrchestrationClient(orchestrationServiceClientMock.Object, durableExtension, httpHandler, new OrchestrationClientAttribute { });
+
+            // This is super hacky, but required due to the circular dependency of ExtendedHttpApiHandler requiring IDurableOrchestrationClient and DurableOrchestrationClient requiring ExtendedHttpApiHandler
+            httpHandler.InnerClient = durableOrchestrationClient;
+
+            string sampleUrl = "https://samplesite.azurewebsites.net";
+            string sampleId = Guid.NewGuid().ToString();
+
+            var netFrameworkRequest = new HttpRequestMessage(HttpMethod.Get, sampleUrl);
+            HttpRequest netCoreRequest = await ConvertHttpRequestMessageAsync(netFrameworkRequest);
+
+            HttpResponseMessage netFrameworkResponse = durableOrchestrationClient.CreateCheckStatusResponse(netFrameworkRequest, sampleId);
+            HttpResponseMessage netCoreResponse = (HttpResponseMessage)((ObjectResult)durableOrchestrationClient.CreateCheckStatusResponse(netCoreRequest, sampleId)).Value;
```

Comment:
>HttpResponseMessage netCoreResponse [](start = 12, length = 35)

Based on the variable name, I would have expected this to be of type `HttpResponse`. Why do we cast it to `HttpResponseMessage`?

Reply:
The easiest way to convert `HttpResponseMessage` into an `IActionResult` (which Controllers in .NET Core implicitly turn into `HttpResponse` I believe) is the following code:

```
var result = new ObjectResult(response);
result.Formatters.Add(new HttpResponseMessageOutputFormatter());
```

With that custom formatter, the Controller logic should automatically apply the `HttpRequestMessage` data to the `HttpResponse`. Since I don't have access to a controller to do that for me, the easiest way to validate is just to make sure that the `HttpResponseMessage` fed into the `ObjectResult` is identical to the `HttpResponseMessage` returned by our old method, and trust that the ` HttpResponseMessageOutputFormatter` does its job correctly.